### PR TITLE
fix: do not render modal at all if no data

### PR DESCRIPTION
**Pull request by apalchys**
**🟢 Add `deploy` label if you want to deploy this Pull Request to staging environment**

#### 🧑‍⚖️ Pull Request Naming Convention

- Title should follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- Do not put issue id in title
- Do not put WIP in title. Use [Draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/) functionality
- Consider to add `area:*` label(s)

* [x] I followed naming convention rules

---

#### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 🔗 Related issue link

#2035 

#### 💡 Background and solution

_Describe the big picture of your changes here_

#### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Database migration is added or not needed
- [x] Documentation is updated/provided or not needed
- [x] Changes are tested locally

### CODE REVIEW

File: client/src/pages/admin/courses.tsx [hunk 1]
```diff
@@ -241,12 +244,12 @@ function Page() {
           label="Minimum Students per Mentor"
           rules={[{ min: 1, type: 'integer', message: 'Ensure that the input, if provided, is a positive integer.' }]}
         >
-          <InputNumber step={1} defaultValue={2} />
+          <InputNumber step={1} />
```
Comment (valerydluski) [inline client/src/pages/admin/courses.tsx:247]: why did you remove the default value here? Does it cause any problems?

**Review by valerydluski — COMMENTED** [review]
Thank you so much for the fix

File: client/src/pages/admin/courses.tsx (hunk 1 above)
Comment (apalchys) [reply client/src/pages/admin/courses.tsx:247]: @valerydluski yes, it conflicts with input mode (controlled)  as per warning. I moved to initial values.
